react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+
+class RegistrationForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      password: ''
+    };
+    this.handleUsernameChange = this.handleUsernameChange.bind(this);
+    this.handlePasswordChange = this.handlePasswordChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleUsernameChange(e) {
+    this.setState({
+      username: e.target.value
+    });
+  }
+
+  handlePasswordChange(e) {
+    this.setState({
+      password: e.target.value
+    });
+  }
+
+  handleSubmit(e) {
+    console.log(this.state);
+    e.preventDefault();
+  }
+
+  render() {
+    return (
+      <form>
+        <label>
+          Username
+        <input value={this.state.username} onChange={this.handleUsernameChange}></input>
+        </label>
+        <label>
+          Password
+          <input value={this.state.password} type="password" onChange={this.handlePasswordChange}></input>
+        </label>
+        <button onClick={this.handleSubmit} type="button">Sign Up</button>
```

Comment:
`handleSubmit` should go on the `<form>` and not the `<button>`

Reply:
copy, thanks